src/nwb_datajoint/common/common_spikesorting.py

```diff
@@ -521,455 +369,208 @@ def get_sort_interval_valid_times(self, key):
-                                        output_folder=os.getenv('MS4_TMP_OUTPUT'),
-                                        **sort_parameters['parameter_dict'])
-        else:
-            sorting = ss.run_sorter(key['sorter_name'], recording,
-                                    output_folder=None,
-                                    **sort_parameters['parameter_dict'])
+        1. Loads saved recording and runs the sort on it with spikeinterface
+        2. Saves the sorting with spikeinterface
+        3. Creates an analysis NWB file and saves the sorting there
+           (this is redundant with 2; will change in the future)
+        """
+        
+        recording_path = (SpikeSortingRecording & key).fetch1('recording_path')
+        recording = si.load_extractor(recording_path)
+        
+        timestamps = SpikeSortingRecording._get_recording_timestamps(recording)
+
+        # load valid times
+        artifact_times = (ArtifactRemovedIntervalList & key).fetch1('artifact_times')
+        if artifact_times.ndim==1:
+            artifact_times = np.expand_dims(artifact_times,0)
+        
+        if artifact_times:
+            # convert valid intervals to indices
+            list_triggers = []
+            for interval in artifact_times:
+                list_triggers.append(np.arange(np.searchsorted(timestamps, interval[0]),
+                                            np.searchsorted(timestamps, interval[1])))
+            list_triggers = np.asarray(list_triggers).flatten().tolist()
+            
+            if recording.get_num_segments()>1:
+                recording = si.concatenate_recordings(recording.recording_list)
+            recording = st.remove_artifacts(recording=recording, list_triggers=list_triggers,
+                                            ms_before=0, ms_after=0, mode='zeros')
+        
+        # NOTE: decided not to do this and instead just use remove_artifacts; keep for now
+        # slice valid times and append them
+        # rec_list = []
+        # for idx_interval in artifact_removed_valid_times_idx_list:
+        #     if recording.get_num_segments()>1:
+        #         segment_ind = np.ravel(np.argwhere(cumsum_frames <= idx_interval[0]))[-1]
+        #         rec = recording.recording_list[segment_ind]
+        #         sliced_rec = rec.frame_slice(start_frame=idx_interval[0]-cumsum_frames[segment_ind],
+        #                                      end_frame=idx_interval[1]-cumsum_frames[segment_ind])
+        #     else:
+        #         sliced_rec = recording.frame_slice(start_frame=idx_interval[0], end_frame=idx_interval[1])
+        #     rec_list.append(sliced_rec)
+        # recording = si.append_recordings(rec_list)
+                
+        preproc_params = (SpikeSortingPreprocessingParameters & key).fetch1('preproc_params')
+        recording = st.preprocessing.whiten(recording=recording, seed=preproc_params['seed'])
```

Comment:
Could you put in a parameter to make whitening skippable?  This would make it easier for the clusterless marks generation

Reply:
now you can skip whitening by setting 'whiten'=False when defining preprocessing parameters